<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>pleioh2g-tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">pleioh2g-tutorial</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pleioh2g)</span></code></pre></div>
<p>pleioh2g-tutorial.R ### <strong>Installation</strong>:</p>
<pre class="&gt;"><code>install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;yjzhao1004/pleioh2g&quot;)
library(pleioh2g)</code></pre>
<div id="steps" class="section level3">
<h3><strong>Steps</strong></h3>
</div>
<div id="step-1-prepare-ldsc-input-format-data-for-multiple-traits." class="section level3">
<h3>Step 1: Prepare LDSC input-format data for multiple traits.</h3>
<p>PHBC needs LDSC .sumstats format data as input, so you first need to
reformat the GWAS summary statistics as LDSC requested. We strongly
recommend that you use the script munge_sumstats.py included in LDSC
python package (<a href="https://github.com/bulik/ldsc" class="uri">https://github.com/bulik/ldsc</a>) to convert your own GWAS
summary statistics into LDSC format. (ref. Bulik-Sullivan et al. 2015b
<em>Nat Genet</em>) We also provide all LDSC-format .sumstat.gz data
used in our analyses. (See Data preparation) * Examples of three
phenotypes (“401.1”, “250.2”, “296.22”) have been implemented in our
package. You can use codes below to reload them.</p>
<pre><code>library(pleioh2g)
munged_sumstats = list(&quot;401.1&quot; = sumstats_munged_example_input(example = &quot;401.1&quot;), &quot;250.2&quot; = sumstats_munged_example_input(example = &quot;250.2&quot;),&quot;296.22&quot; = sumstats_munged_example_input(example = &quot;296.22&quot;))</code></pre>
</div>
<div id="step-2-compute-pleiotropic-heritability-with-bias-correction" class="section level3">
<h3>Step 2: Compute pleiotropic heritability with bias correction</h3>
<p>The function <strong>pruning_pleioh2g_wrapper()</strong> (See example
as below) is to compute h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP>
while performing pruning and bias correction with ldsc-format GWAS
summary statistics (.sumstat.gz) as input.</p>
<ul>
<li>We note that h<SUP>2</SUP><SUB>pleio</SUB> is a function of both the
target disease and the selected set of auxiliary diseases/traits. We use
the ratio of pleiotropic heritability vs. total heritability
(h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP>) to quantify the
proportion of genetic variance that is pleiotropic.</li>
<li>We just use 5 jackknife blocks and 3 traits as example for quick
computation test. If you set 200 jackknife blocks and include more than
50 traits, the procedure will cost more than 10 hours.</li>
</ul>
<pre><code># Specify phenotype names
phenotype&lt;-c(&quot;401.1&quot;,&quot;250.2&quot;,&quot;296.22&quot;)

# First to determine which disease in your list is the target disease
G = 1 # Index of target disease in trait list - this example is to compute pleiotropic heritability for &quot;401.1&quot;.

# Input ldsc format .sumstat.gz data
munged_sumstats = list(&quot;401.1&quot; = sumstats_munged_example_input(example = &quot;401.1&quot;), &quot;250.2&quot; = sumstats_munged_example_input(example = &quot;250.2&quot;),&quot;296.22&quot; = sumstats_munged_example_input(example = &quot;296.22&quot;))

# Specify reference LD data: ld and wld path; and hapmap 3 SNPs list
ld_path&lt;-fs::path(fs::path_package(&quot;extdata/eur_w_ld_chr&quot;, package = &quot;pleioh2g&quot;))
wld_path&lt;-fs::path(fs::path_package(&quot;extdata/eur_w_ld_chr&quot;, package = &quot;pleioh2g&quot;))
hmp3&lt;-fs::path(fs::path_package(&quot;extdata/w_hm3.snplist&quot;, package = &quot;pleioh2g&quot;))

# If you trait is disease phenotype or the other binary trait, specify prevalence to compute the liability-scale heritability; If you don&#39;t specify this, it will compute observed-scale heritability.
sample_prev &lt;- c(0.37,0.1,0.17)
population_prev &lt;- c(0.37,0.1,0.17)

# Specify number of genomic-jackknife block; We use n_block = 5 as example for quick computation, but we recommand to use 200 jackknife blocks; If you don&#39;t specify this, the default number is 200.
n_block&lt;-5
 
# Specify number of Monte Carlo sampling iterations in bias correction; If you don&#39;t specify this, the default number is 1000.
sample_rep &lt;- 1000 

post_correction_results&lt;-pruning_pleioh2g_wrapper(G,phenotype,munged_sumstats,ld_path, wld_path, sample_prev, population_prev,n_block, hmp3,sample_rep)</code></pre>
<p>An output line will provide your post-correction
h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP> estimate, along with a
result list <code>post_correction_results</code>, containing the
following elements： - <code>target_disease</code> (character): The
value “401.1”. - <code>target_disease_h2_est</code> (numeric): target
disease h<SUP>2</SUP>. - <code>target_disease_h2_se</code> (numeric):
target disease h<SUP>2</SUP> s.e.. - <code>selected_auxD</code>
(character): auxiliary diseases. - <code>h2pleio_uncorr</code>
(numeric): pre-correction h<SUP>2</SUP><SUB>pleio</SUB> estimate. -
<code>h2pleio_uncorr_se</code> (numeric): pre-correction
h<SUP>2</SUP><SUB>pleio</SUB> jackknife s.e. estimate. -
<code>percentage_h2pleio_uncorr</code> (numeric): pre-correction
h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP> estimate. -
<code>percentage_h2pleio_uncorr_se</code> (numeric): pre-correction
h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP> jackknife s.e. estimate. -
<code>percentage_h2pleio_jackknife_uncorr</code> (numeric): vector of
all pre-correction h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP>
jackknife estimates in default 200 blocks. - <code>h2pleio_corr</code>
(numeric): post-correction h<SUP>2</SUP><SUB>pleio</SUB> estimate. -
<code>h2pleio_corr_se</code> (numeric): post-correction
h<SUP>2</SUP><SUB>pleio</SUB> jackknife s.e. estimate. -
<code>percentage_h2pleio_corr</code> (numeric): post-correction
h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP> estimate. -
<code>percentage_h2pleio_corr_se</code> (numeric): post-correction
h<SUP>2</SUP><SUB>pleio</SUB> / h<SUP>2</SUP> jackknife s.e. estimate. -
<code>corrected_weight</code> (numeric): corrected weight ξc in bias
correction.</p>
</div>
<div id="leave-category-out-analysis-instruction" class="section level3">
<h3>Leave-category-out analysis instruction</h3>
<ul>
<li>If you want to compute pleiotropic heritability with respect to a
subset of auxiliary disease, you can just specify the subset as the
auxiliary disease and repeat step 1 and 2</li>
<li>Estimating the contribution of a subset of auxiliary categories or
diseases requires careful consideration due to an inherent pruning
procedure. Our recommended approach is as follows.
<ul>
<li>For analyses excluding the target disease category or a single
specified category:
<ul>
<li>First, conduct the analyses using all auxiliary diseases.</li>
<li>Then, remove the auxiliary diseases within the specified category
from <code>post_correction_results$selected_auxD</code>.</li>
<li>Update the <code>phenotype</code> and <code>munged_sumstats</code>
input parameters with the remaining auxiliary diseases.</li>
<li>Rerun Step 2.</li>
</ul></li>
<li>For analyses excluding the target disease category and an additional
specified category:
<ul>
<li>First, conduct the analyses using all auxiliary diseases
<em>excluding</em> the target disease category.</li>
<li>Then, remove the auxiliary diseases within the specified category
from <code>post_correction_results$selected_auxD</code>.</li>
<li>Update the <code>phenotype</code> and <code>munged_sumstats</code>
input parameters with the remaining auxiliary diseases.</li>
<li>Rerun Step 2.</li>
</ul></li>
<li>This procedure will ensure the consistency of pruning with the
subsequent analyses.</li>
</ul></li>
</ul>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
